docs/mathematics-algorithms/fermat's-Little-Theorem.md

```diff
@@ -0,0 +1,174 @@
+---
+id: fermat's-Little-theorem
+title: Fermat's Little Theorem
+sidebar_label: Fermat's Little Theorem MMI
+sidebar_position: 5
+description: A comprehensive guide to calculating the modular multiplicative inverse using Fermat's Little Theorem.
+tags: [number theory, modular arithmetic, cryptography,competitive progrmaing]
+---
+
+## Modular Multiplicative Inverse Using Fermat's Little Theorem
+
+**Definition:**
+
+The modular multiplicative inverse of an integer `a` modulo `m` is an integer `b` such that `a * b ≡ 1 (mod m)`. In other words, `b` is the inverse of `a` in the ring of integers modulo `m`.
+
+**Fermat's Little Theorem for Modular Inverses:**
+
+Fermat's Little Theorem provides a direct way to calculate the modular multiplicative inverse of `a` modulo `p` when `p` is a prime number. It states that:
+a^(p-2) ≡ a^(-1) (mod p)
+This means that raising `a` to the power of `p-2` modulo `p` gives you the modular multiplicative inverse of `a`.
+<!-- 
+### Remember
+(A/B)%mod != ((A%mod)/(B%mod))%mod why ? do experiments :) -->
```

Comment:
remove comment

Reply:
Check both which is better ALL DONE or Changes Done